lib/plug_caisson.ex

```diff
@@ -13,7 +13,9 @@ defmodule PlugCaisson do
   @callback init(opts :: term()) :: {:ok, state :: term()} | {:error, term()}
   @callback deinit(state :: term()) :: term()
   @callback process(state :: term(), data :: binary(), opts :: keyword()) ::
-              {:ok, binary()} | {:more, binary()} | {:error, term()}
```

Comment:
would need documentation around how custom algos handle state, as well as example of what it is usually used for.

Reply:
Added documentation and fixed some issues related to the documentation.